url71665d7d68_branscha_lib_scripty_tgz-pJ8-com_sdicons_scripty_ScriptyAutoReplJ8/src/com/sdicons/scripty/spec/args/ArgListBuilderUtil.java

```
@@ -0,0 +1,714 @@
+        throws CommandException
+        {
+            try
+            {
+                NOARGS.guard(aArgs, aCtx);
+                return BIGDECIMAL_TUPLE;
+            }
+            catch (ArgSpecException e)
+            {
+                throw new CommandException(e.getMessage());
+            }
+        }
+
+        @ScriptyCommand(name="Integer")
+        public static TypeTuple integerType(Object[] aArgs, IContext aCtx)
+        throws CommandException
+        {
+            try
+            {
+                NOARGS.guard(aArgs, aCtx);
+                return INTEGER_TUPLE;
+            }
+            catch (ArgSpecException e)
+            {
+                throw new CommandException(e.getMessage());
+            }
+        }
+        
+        @ScriptyCommand(name="BigInteger")
+        public static TypeTuple bigIntegerType(Object[] aArgs, IContext aCtx)
+        throws CommandException
+        {
+            try
+            {
+                NOARGS.guard(aArgs, aCtx);
+                return BIGINTEGER_TUPLE;
+            }
+            catch (ArgSpecException e)
+            {
+                throw new CommandException(e.getMessage());
+            }
+        }
+        
+        @ScriptyCommand(name="Boolean")
+        public static TypeTuple booleanType(Object[] aArgs, IContext aCtx)
+        throws CommandException
+        {
+            try
+            {
+                NOARGS.guard(aArgs, aCtx);
+                return BOOLEAN_TUPLE;
+            }
+            catch (ArgSpecException e)
+            {
+                throw new CommandException(e.getMessage());
+            }
+        }
+        
+        @ScriptyCommand(name="Byte")
+        public static TypeTuple byteType(Object[] aArgs, IContext aCtx)
+        throws CommandException
+        {
+            try
+            {
+                NOARGS.guard(aArgs, aCtx);
+                return BYTE_TUPLE;
+            }
+            catch (ArgSpecException e)
```

Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)